### PR TITLE
Logs Panel: Table UI - Add time and body fields to column selection in logs table visualization

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
Adding the time and body (line) fields to logs panel table in explore.

Time and body fields are now in column selection UI, default to both being selected, and
image

Active fields will display before inactive fields
image

Inactive fields will sort by density (percent of log lines in sample this label exists on)
image

**Why do we need this feature?**
The log line body field on its own can be very long, and not useful for visualizing, especially in a table with narrow columns.
Gives users more customization over logs table output in explore.

**Who is this feature for?**
Dmytro Sychevsky, and other early users/adopters of the experimental logs table in explore.

**Special notes for your reviewer:**
Now the user can "unselect" all columns, for now the table will just automatically re-select the time/body fields whenever we get a table with no columns (the initial state).

It feels odd when you unselect all columns and it bounces back to selecting two columns. Maybe the last column selection input is disabled so you can't select 0 columns?

Please check that:
- [ ] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/explore/Logs/LogsTableNavColumn.tsx [hunk 1]
```diff
@@ -45,17 +45,76 @@ export const LogsTableNavColumn = (props: {
...
-              className={styles.checkbox}
-              label={labelName}
-              onChange={() => toggleColumn(labelName)}
-              checked={labels[labelName]?.active ?? false}
-            />
-            <span className={styles.labelCount}>({labels[labelName]?.percentOfLinesWithLabel}%)</span>
-          </div>
-        ))}
+        {labelKeys
+          .sort((a, b) => {
```
Comment (gtk-grafana) [inline public/app/features/explore/Logs/LogsTableNavColumn.tsx:49]: Pull this out and refactor this mess please

**Comment by svennergr** [comment]
> Inactive fields will sort by density (percent of log lines in sample this label exists on)

Is this based on some feedback? For me this makes it very unintuitive and hard to look for a certain label, because I don't know the density. I think alphabetically might make more sense here. WDYT?

**Comment by gtk-grafana** [comment]
@svennergr good call, I've changed the sort to alphabetic instead of by density

File: public/app/features/explore/Logs/LogsTable.tsx [hunk 2]
```diff
@@ -48,6 +48,9 @@ export function LogsTable(props: Props) {
 
   const prepareTableFrame = useCallback(
     (frame: DataFrame): DataFrame => {
+      if (!frame || !frame.length) {
+        return frame;
+      }
```
Comment (svennergr) [inline public/app/features/explore/Logs/LogsTable.tsx:53]: Given the type `frame` should always be trueish, so I think that can be reduced to 

```suggestion
      if (!frame.length) {
        return frame;
      }
```

**Review by svennergr — APPROVED** [review]
Just a small nit, rest LGTM!

**Pull request merged**